src/raps/common.ts

```diff
@@ -139,24 +161,30 @@ const executeAction = async (
       parameters,
       baseNonce
     );
-    return nonce;
+    return { baseNonce: nonce, errorMessage: null };
   } catch (error) {
-    logger.sentry('[3 INNER] error running action');
+    logger.sentry('[3 INNER] error running action, code:', error?.code);
     captureException(error);
     analytics.track('Rap failed', {
       category: 'raps',
       failed_action: type,
       label: rapName,
     });
-    return null;
+    // If the first action failed, return an error message
+    if (index === 0) {
```

Comment:
I don't care about parsing errors after the first rap as everything else is in the background task